src/vs/base/common/async.ts

```diff
@@ -94,18 +94,21 @@ export function raceCancellationError<T>(promise: Promise<T>, token: Cancellatio
 }
 
 /**
- * Returns as soon as one of the promises is resolved and cancels remaining promises
+ * Returns as soon as one of the promises resolves or rejects and cancels remaining promises
  */
 export async function raceCancellablePromises<T>(cancellablePromises: CancelablePromise<T>[]): Promise<T> {
```

Comment:
I think this is an improvement, thanks, but it is a behavior change so make sure you take a look at the other places it's used and try to check whether that might be an issue

Reply:
Yeah it's only used in [one other place](https://github.com/microsoft/vscode/blob/6171ba5a8bf55b91954c6d121a920fa9f83bff29/src/vs/workbench/contrib/extensions/browser/extensionRecommendationNotificationService.ts#L238-L241) and I think based on the code it seems ok. @sandy081 FYI